pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -264,7 +264,7 @@ func newNSTemplateTiers(decoder runtime.Decoder, namespace string, templatesByTi
 //       template: >
 //         <yaml-ns-template>
 // ------
-func newNSTemplateTier(decoder runtime.Decoder, namespace, tier string, tmpls templates) (*toolchainv1alpha1.NSTemplateTier, error) {
+func newNSTemplateTier(namespace, tier string, tmpls templates) (*toolchainv1alpha1.NSTemplateTier, error) {
```

Comment:
wouldn't it be better to drop the `templates` type and replace it with a list of `TierTemplates`? Then, you could take the actual `TemplateRef` at line 291 instead of constructing the ref again.
I have kind of feeling that the function(s) is still using the previous style but creating a new format. But it's maybe just my feeling/opinion.

Reply:
that's a good point! let me check that ;)

Reply:
@MatousJobanek done in e28ea1b but it may be out-of-scope for this PR so I can address it separately if it makes more sense or if it can simplify the review of this PR ;)